Sources/PostgreSQL/Connection/PostgreSQLConnection.swift

```diff
@@ -134,8 +134,16 @@ public final class PostgreSQLConnection {
         }
     }
 
+    internal var beforeClose: ((PostgreSQLConnection) -> Future<Void>)?
```

Comment:
In case this gets set by two different callers, some resources might not get cleaned up. How about making this an array of close handlers that you can append to? (Just a suggestion.)